lib/assert.js

```diff
@@ -15,13 +15,16 @@ const noop = () => {};
 Object.defineProperty(x, 'AssertionError', {value: assert.AssertionError});
 
 function create(val, expected, operator, msg, fn) {
-	return {
-		actual: val,
-		expected,
-		message: msg,
-		operator,
-		stackStartFunction: fn
-	};
+	if (msg === undefined || typeof msg === 'string') {
+		return {
+			actual: val,
+			expected,
+			message: msg,
+			operator,
+			stackStartFunction: fn
+		};
+	}
+	throw new TypeError('assertion message must be a string');
```

Comment:
Would be better to invert the check and put the throw in the if-statement. Also: `assertion` => `Assertion`